core/include/userver/engine/single_waiting_task_mutex.hpp

```diff
@@ -0,0 +1,84 @@
+#pragma once
```

Comment:
issue(non-blocking): The sole purpose of `SingleWaitingTaskMutex` is performance, but there are no benchmarks for it.